src/main/cpp/src/hive_hash.cu

```diff
@@ -203,18 +203,64 @@ class hive_device_row_hasher {
     {
     }
 
-    template <typename T, CUDF_ENABLE_IF(not cudf::is_nested<T>())>
-    __device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
-                                            cudf::size_type row_index) const noexcept
-    {
-      return this->hash_functor.template operator()<T>(col, row_index);
-    }
+    struct stack_element {
+      cudf::column_device_view col;
+      int child_index;
+      hive_hash_value_t hash;
+
+      __device__ stack_element() = delete;
+      __device__ stack_element(cudf::column_device_view col) : col(col), child_index(0), hash(HIVE_INIT_HASH) {}
+    };
 
-    template <typename T, CUDF_ENABLE_IF(cudf::is_nested<T>())>
+    template <typename T>
```

Comment:
We'd better keep the original two methods here to separate the non-nested types from nested types. It can avoid some useless "if" condition check and "accumulate" execution. Which will impact the perf when there are more than millions of rows.